pymc/distributions/ordered.py

```diff
@@ -0,0 +1,133 @@
+            shape = (*tuple(size), core_shape)
+            dist = change_dist_size(dist, shape)
+
+        sorted_rv = pt.sort(dist, axis=-1)
+
+        return OrderedRV(
+            inputs=[dist],
+            outputs=[sorted_rv],
+        )(dist)
+
+
+class Ordered(Distribution):
+    r"""Univariate IID Ordered distribution.
+
+    The pdf of the oredered distribution is
+
+    .. math::
+        f(x_1, ..., x_n) = n!\prod_{i=1}^n f(x_{(i)}),
+        where x_1 <= x2 <= ... <= x_n
+
+    Parameters
+    ----------
+    dist: unnamed_distribution
+        Univariate IID distribution which will be sorted.
+
+        .. warning:: dist will be cloned, rendering it independent of the one passade as input
+
+    Examples
+    --------
+    .. code-block:: python
+        import pymc as pm
+
+        with pm.Model():
+            x = pm.Normal.dist(mu=0, sigma=1)  # Must be IID
+            ordered_x = pm.Ordered("ordered_x", dist=x, shape=(3,))
+
+        pm.draw(ordered_x, random_seed=52)  # array([0.05172346, 0.43970706, 0.91500416])
+    """
+
+    rv_type = OrderedRV
+    rv_op = OrderedRV.rv_op
+
+    def __new__(cls, name, dist, *, support_shape=None, **kwargs):
+        support_shape = get_support_shape_1d(
+            support_shape=support_shape,
+            shape=None,  # shape will be checked in `cls.dist`
+            dims=kwargs.get("dims", None),
+            observed=kwargs.get("observed", None),
+        )
+        return super().__new__(cls, name, dist, support_shape=support_shape, **kwargs)
+
+    @classmethod
+    def dist(cls, dist, *, support_shape=None, **kwargs):
+        if not isinstance(dist, TensorVariable) or not isinstance(
+            dist.owner.op, RandomVariable | SymbolicRandomVariable
+        ):
+            raise ValueError(
+                f"Ordered dist must be a distribution created via the `.dist()` API, got {type(dist)}"
+            )
+        if dist.owner.op.ndim_supp > 0:
+            raise NotImplementedError("Ordering of multivariate distributions not supported")
+        if not all(
+            all(param.type.broadcastable) for param in dist.owner.op.dist_params(dist.owner)
+        ):
+            raise ValueError("Ordered dist must be an IID variable")
+
+        support_shape = get_support_shape_1d(
+            support_shape=support_shape,
+            shape=kwargs.get("shape", None),
+        )
+        if support_shape is not None:
+            dist = change_dist_size(dist, support_shape)
+
+        dist = pt.atleast_1d(dist)
+
+        return super().dist([dist], **kwargs)
+
+
+@_default_transform.register(OrderedRV)
+def default_transform_ordered(op, rv):
+    if rv.type.dtype.startswith("float"):
+        return ordered
+    else:
+        return None
+
+
+@_support_point.register(OrderedRV)
+def support_point_ordered(op, rv, dist):
+    # FIXME: This does not work with the default ordered transform
+    #  It leads to a -inf log_jac_det
+    # return support_point(dist)
```

Comment:
@aseyboldt @lucianopaz any idea if we could easily modify the OrderedTransform to work nicely with equal values in the constrained space? The current one maps a `[0, 0, 0]` vector to `[0, -inf, -inf]`. My guess is not. But then I don't have a nice way to define support_point. I could try to add a jitter but that will fail for bounded distributions.